### DISCUSSION TITLE
No-downtime upgrades

### DISCUSSION THREAD

**Discussion by lortabac** [ory/hydra#3118] [Q&A] [answered]
Is it possible to upgrade Hydra with zero downtime?
In particular, can a new version run safely on a database that hasn't been migrated yet? Or vice versa, can an old version run on a database that is on a more recent version (within a reasonable distance)?

**Comment by vinckr** [comment]
>can a new version run safely on a database that hasn't been migrated yet?

I never tried this out but my understanding is that you need to run [`hydra-migrate-sql`](https://www.ory.sh/docs/hydra/cli/hydra-migrate-sql) before. 
Is that correct or can you add something regarding no-downtime upgrading @grantzvolsky ?

**Comment by grantzvolsky** [comment, marked as answer]
It is generally not possible to upgrade hydra with zero downtime. See https://github.com/ory/hydra/issues/1236#issuecomment-878584740
